Jenkinsfile

```diff
@@ -271,6 +271,30 @@ def getuid() {
     return cached_uid
 }
 
+// Default priority is 3, lower is better.
+// The parameter for a job is set using the script/Jenkinsfile from the
+// previous build of that job, so the first build of any PR will always
+// run at default because Jenkins sees it as a new job, but subsequent
+// ones will use the value from here.
+// The advantage therefore is not to change the priority of PRs, but to
+// change the master branch itself to run at lower priority, resulting
+// in faster time-to-result for PRs.
+
+String get_priority() {
+    node {
+        checkoutScm()
+        if (cachedCommitPragma(pragma: 'Queue-jump', def_val: false)) {
+	    string p = '2'
+	} else if (env.BRANCH_NAME == 'master') {
+	    string p = '4'
+	} else {
+	    string p = ''
```

Comment:
Is there a reason this is '' rather than '3'?   I guess if 3 is the default, it will end up as 3 but I don't see a reason to not set it such here unless we want to change the defaults?

Reply:
This was at the request of @brianjmurrell

Reply:
> Is there a reason this is '' rather than '3'? I guess if 3 is the default, it will end up as 3 but I don't see a reason to not set it such here unless we want to change the defaults?

`''` means to use the default defined in the Jenkins configuration.  This allows us to control the default without having to push PRs.